### DISCUSSION TITLE
Behavior of the dropdown list in ingredients selection

### DISCUSSION THREAD

**Discussion by rkuzmenkov** [TandoorRecipes/recipes#2963] [Q&A]
I'm looking for some time for a self-hosted recipe organizer. And for me, Tandoor looks better than its competitors.
But the behavior of the ingredient dropdown list when I type the name of it is a bit illogical to me.

For example, I want to add some 'asparagus' to the list.
I type 'aspar'. And the first option is to create a new ingredient until I type the whole name. So every second time I finish creating a new ingredient when not typing a full name
![tandoor_ingredients](https://github.com/TandoorRecipes/recipes/assets/117346619/a890a468-4f84-4686-801b-7aca941e8b2b)

I'd expect the following:
* Names that don't match the typed part are filtered out completely.
* Names that match are on top of the list. So if you press 'Tab' or 'Enter' the first element of the list is selected.
* The typed part of the name is at the bottom of the list. If you select it then a new entry is created.

Is there an option to change the behavior of the dropdown list?
Or am I the only one who thinks it's not convenient?

**Comment by smilerz** [comment]
You can turn off fuzzy match in search settings to only return partial matches.
![image](https://github.com/TandoorRecipes/recipes/assets/1808639/61a01472-29e6-48f2-9c4d-dd82dfe5289c)

Tab selection and putting create at the bottom are not possible.

**Reply by rkuzmenkov** [reply]
Thank you for a tip with Fuzzy Lookups.

But the rest is a showstopper for me. I'd really like to have it configurable.

**Comment by rkuzmenkov** [comment]
This also looks unexpected to me.
Why does it show me the ingredients I just added to the list? Even if they don't match.
I doubt I need to add one ingredient twice per recipe.
![tandoor_ingredients_2](https://github.com/TandoorRecipes/recipes/assets/117346619/7ea84fc6-61f4-4e18-97b6-69b51ef2cdd8)
